hr_holidays_credit/models/hr_leave_type.py

```diff
@@ -6,70 +6,86 @@
     )
     creditable_employee_category_ids = fields.Many2many(
-        string='Creditable Employee Tags',
-        comodel_name='hr.employee.category',
+        string="Creditable Employee Tags",
+        comodel_name="hr.employee.category",
         help=(
-            'If set, limits credit allowance to employees with at least one of'
-            ' specified tags'
+            "If set, limits credit allowance to employees with at least one of"
+            " specified tags"
         ),
     )
     creditable_department_ids = fields.Many2many(
-        string='Creditable Departments',
-        comodel_name='hr.department',
+        string="Creditable Departments",
+        comodel_name="hr.department",
         help=(
-            'If set, limits credit allowance to employees of specified'
-            ' departments'
+            "If set, limits credit allowance to employees of specified" " departments"
         ),
     )
 
     @api.multi
     def name_get(self):
-        context_employee_id = self._context.get('employee_id')
+        context_employee_id = self._context.get("employee_id")
 
         res = []
         for record in self:
             record_name = record.name
 
             extra = None
-            if record.allocation_type != 'no' and context_employee_id:
-                if record.virtual_remaining_leaves >= 0:
+            if record.allocation_type != "no" and context_employee_id:
+                amount = (
+                    float_round(record.virtual_remaining_leaves, precision_digits=2)
+                    or 0.0
+                )
+                if amount >= 0:
+                    if record.request_unit == "day":
+                        extra = (
+                            _("%g day available") % amount
+                            if abs(amount) <= 1
+                            else _("%g days available") % amount
+                        )
+                    elif record.request_unit == "hour":
+                        extra = (
+                            _("%g hour available") % amount
+                            if abs(amount) <= 1
+                            else _("%g hours available") % amount
+                        )
+
                     if record.allow_credit:
-                        extra = _('%g available + credit')
-                    else:
-                        extra = _('%g available')
-                    extra = extra % (
-                        float_round(
-                            record.virtual_remaining_leaves,
-                            precision_digits=2
-                        ) or 0.0,
-                    )
-                elif record.allow_credit:
-                    extra = _('%g used in credit') % (
-                        float_round(
-                            -record.virtual_remaining_leaves,
-                            precision_digits=2
-                        ) or 0.0,
-                    )
+                        extra += " " + _("+ credit")
```

Comment:
Don't do this, it's incorrect for i18n to "construct" sentences. You have to localize either entire message or major part of it.